a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -683,18 +683,24 @@ class AllocateTrunk(BaseNetworkTask):
 
     def execute(self, vip):
         parent_port_id = vip.port_id
-        LOG.debug('Creating trunk for port with ID: {}'.format(parent_port_id))
+        LOG.debug('Creating trunk for port with ID: %s', parent_port_id)
         self.network_driver.allocate_trunk(parent_port_id)
 
-    def revert(self, result, *args, **kwargs):
-        pass 
+    def revert(self, result, vip, *args, **kwargs):
+        try:
+            parent_port = self.network_driver.get_plugged_parent_port(vip)
+            self.network_driver.deallocate_trunk(parent_port.trunk_id)
+        except Exception:
+            LOG.warning('Failed to deallocate trunk port: %(trunk)s '
```

Comment:
Just for precaution - not a change. do we get this trunk variable in args ?

Reply:
No I only have the parent port being passed around since it can be used to expose everything else.